### PR TITLE
adds cloneloss to robots and makes them health scannable

### PR DESCRIPTION
**Pull request by Sebbe9123**
## About The Pull Request
This PR adds Integrity to robot equivilant to cloneloss
Psydrain gives robots cloneloss.
Robots can now be scanned by health analyzers.
Adds automatic "cryotube" called robotic cradle, fully heals a robot of all ailments except death, including limbs.
## Why It's Good For The Game
"Permanent" downsides to dying good, makes robots less able to run in and get revived without consequence.
## Changelog
:cl:
add: Robots now get cloneloss the same as marines
add: Robotic cradle. A shipside machine that fully restores a robot.
balance: Robots can now be health analyzed.
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Magic paste lame. Its just another thing you can carry in the field to not worry about lasting damage.

Add like a robotics maintenance cradle to shipside or something.

**Comment by Sirsnek15** [comment]
The fact that defib automatically heals them to reviveable state is absolutely ridiculous, and this PR is simply compensating for that instead of adressing the problem. This is a very not good idea

**Comment by ilikesaying** [comment]
> The fact that defib automatically heals them to reviveable state is absolutely ridiculous, and this PR is simply compensating for that instead of adressing the problem. This is a very not good idea

as opposed to synths/engineers getting solders to freely repair combat robots/synths?

**Comment by QualityVan** [comment]
I'm not convinced that a separate damage type is justified here when you could coopt cloneloss as robots don't use it. Cocoons dealing periodic damage should probably also be a separate PR.

**Comment by Lumipharon** [comment]
What QV said. This is literally cloneloss with a new name. So just use cloneloss, and make it show up as a different name for robos.
Not atomising is guaranteeing a slow and painful process to merge.

**Comment by Sebbe9123** [comment]
> What QV said. This is literally cloneloss with a new name. So just use cloneloss, and make it show up as a different name for robos. Not atomising is guaranteeing a slow and painful process to merge.

Yup, working on it and the cryotube equivalent, removing the nanopaste.

**Comment by coldironwarrior** [comment]
probably not major but robot sprites have damage states for the brute/burn damages. you cant read their damage levels with a scanner.
having a secret third type of damage that you cant gauge in any manner but just have to /know/ about seems like bad design, unintuitive.

**Comment by AusarAtkin** [comment]
Robot can still sleep, so they dont really lose the groundside option to heal cloneloss. But maybe check if it really heals cloneloss on robot like on human.

**Comment by Aporhtonoma** [comment]
robots are chem immune,crit immune,pain immune,more flat hp having,infinite healing supply having,no perma injuries sustained,no blood losing,no organ damage available AND able to die a infinite amount of times due to no cell damage available. The fact that robots can now finnaly get punished for dying 15 times like marines is good.They have many good things going for them but immunity to death consequenses should not be one of them.

personally I think that a paste that can be carried groundside is still kinda way to strong since you can just bring it with you and it should be like rines where they have a dedicated machine to heal .(which could be added to engieneering both planet and shipside and the coocoon buff is ... questionable . but its a good start

**Comment by TiviPlus** [comment]
you need to update pr body btw

**Comment by ClosetedSkeletonGH** [comment]
Hate to be that person but 90% of salt against robots is just bad/newer xeno’s who don’t understand how robots health work expecting them to go down as soon their health bar hits 0% cause of the weird robo crit mechanic.

Having their health bar fixed to just be full health to no health would be a notable buff to bad/new xeno’s and would widely stop the complaints

**Comment by AlexCrimson** [comment]
Isnt the risk already from the fact its a slow psy gain over time, that can be stopped by Marines recovering the body? Bodies are almost never cocooned for that exact reason. Unless there is a big wipe, or bodies in a very hard-to-reach area.

**Comment by Sebbe9123** [comment]
There, pretty much the final time I'm gonna touch the PR unless there's somehow some bugs in it. If someone wants to code in a cooler way to heal cloneloss, feel free to take the branch and do it. Lost interest. I do have some sprites if you want to do lumi's idea of whatever they wanted, I did cryo and it worked fine but it's up to you.

**Comment by Pixelknightly** [comment]
Isn't nanopaste pretty limited in comparison to cryo? Also, this'll lead to medbay getting looted by robots for all the nanopaste, and it'll just end up not really affecting anything since nanopaste is an item you can bring groundside.

I'm also just not sure this is necessary or even good for the game.

**Comment by Pixelknightly** [comment]
> robots are chem immune,crit immune,pain immune,more flat hp having,infinite healing supply having,no perma injuries sustained,no blood losing,no organ damage available AND able to die a infinite amount of times due to no cell damage available. The fact that robots can now finnaly get punished for dying 15 times like marines is good.They have many good things going for them but immunity to death consequenses should not be one of them.
> 
> personally I think the coocoon buff is ... questionable . but its a good start and i trully hope this gets merged

Are you implying that human marines don't have infinite healing? With the amount of meds most of them carry? Hell, it even happens while they're fighting. Robots can't do that. They gotta stay still and vulnerable. Chem immunity is a double-edged sword. Sentinel neurotox still damages them and defiler gas still blocks their sight. These traits aren't overpowered in the slightest.

They're pain immune, but they're still affected by slowdown. Which is what pain is most useful for.
Also, you're severely underestimating just how slow robots are in comparison to humans. Robots cannot chase effectively, they just can't. They can't flee effectively either, or really do anything with any sense of urgency. So boiler gas clouds have a tendency of fucking them over courtesy of the complete inability to get out of there.

You literally can't overstate the importance of sprinting, and how the lack thereof is a huge negative to combat robots.

**Comment by ilikesaying** [comment]
combat robots are not affected by PAIN but are still AFFECTED by **_DAMAGE SLOWDOWN_**, and their lack of any passive healing options means they are extremely vulnerable to Hit-And-Run attacks, they are especially vulnerable to Sentinel's toxic spit due to it causing Damage-over-Time applied to every limb.

**Comment by Sirsnek15** [comment]
I will say it again: If robots didn't have such a ridiculous revive, no one would have felt compelled to make this, and specifically **no one would have felt that they could just jump back into combat after dying**
Because without the super healing defib, they really wouldn't be able to jump into combat after dying, like they can right now. 

Source: Occasional Robo player

**Comment by Pixelknightly** [comment]
> I will say it again: If robots didn't have such a ridiculous revive, no one would have felt compelled to make this, and specifically **no one would have felt that they could just jump back into combat after dying** Because without the super healing defib, they really wouldn't be able to jump into combat after dying, like they can right now.
> 
> Source: Occasional Robo player

That's **not** a thing they can do. The moment they get revived, it just puts them in a **near-death state**. They still need to spend time fixing themselves if they want to actually fight without dying from a single hit. (And this is assuming their arm isn't broken, which would require them to fix themselves nearly up to full if they actually want to wield their gun.)

The amount of time they spend fixing themselves is **longer** than the amount of time a marine with meraderm, inap, and tricord heals up to full. (I would know, I've seen it happen more than a hundred times.)

You're asking for combat robots to get completely fucked if you remove the medic's ability to just revive them, considering how few of them actually carry around the tools required to repair robots. And no, engineers do not give a shit. They never do. If they did, we would still be seeing a lot of dispensers despite the nerfs. 

You're complaining about an issue that doesn't exist. If anything, the only problem combat robots **had** was the fact that they were able output full dps up until they die. (Which stopped being true when they stopped having unbreakable arms.)

**Comment by Sirsnek15** [comment]
> 

Being revived first try despite any amount of damage is LITERALLY A THING THEY CAN DO. Regardless of what healing is needed afterwards, this is ridiculous. Not as ridiculous as it would be for humans to have this ability, but that doesn't give the current state of things any good reason to be the way they are. Robos go down slow, they get up slow. I'm asking for Robos to require repairs according to the amount of damage they have recieved. In other words, I'm asking for a reasonable approach instead of compounding ridiculous developments.

I'm complaining about the fact that they do not have to be repaired into a reviveable state. **This is an issue that exists**. Its absolutely ridiculous and its forcing people to think that this PR is a good idea. We had solutions for this prior, such as robos helping robos, and it worked alright. To say that they had no or not enough support is factually incorrect.

**Comment by Pixelknightly** [comment]
> > 
> 
> Being revived first try despite any amount of damage is LITERALLY A THING THEY CAN DO. Regardless of what healing is needed afterwards, this is ridiculous. Not as ridiculous as it would be for humans to have this ability, but that doesn't give the current state of things any good reason to be the way they are. Robos go down slow, they get up slow. I'm asking for Robos to require repairs according to the amount of damage they have recieved. In other words, I'm asking for a reasonable approach instead of compounding ridiculous developments.
> 
> I'm complaining about the fact that they do not have to be repaired into a reviveable state. **This is an issue that exists**. Its absolutely ridiculous and its forcing people to think that this PR is a good idea. We had solutions for this prior, such as robos helping robos, and it worked alright. To say that they had no or not enough support is factually incorrect.

Did you misunderstand what I wrote? Yes, being revived immediately is a thing they can do. **They cannot immediately go back to fight however, because that is a dumb fucking idea that will kill them in a single hit.** They have to repair themselves very slowly in comparison to human marines, before they can actually fight.

Robots helping robots working _alright_ is pretty fucking shit in comparison to human marines having an entire god damn support system. Especially since the removal of solder. Y'know, the thing that got removed, so they gave robots the compensation of being immediately revived?

I mean, did you forget how many people complained about the removal of solder because as it turns out, being revived immediately isn't a good replacement for the ability to always be able to repair robots? And not run out of resources every ten seconds?

What exists right now **IS** the reasonable approach. **It doesn't matter if a robot gets revived immediately.** **They still have to spend the next one and a half minutes pressing a single button to repair themselves to full.** Otherwise a single boiler cloud could kill them.

If we made changes based entirely on people thinking other people's abilities are bullshit, stuff like jetpacks and crusher fling going over cades would've been removed ages ago. If those aren't a problem, why is this? Especially with the two huge negatives that already exist for robots?

**Comment by AusarAtkin** [comment]
> > 
> 
> Being revived first try despite any amount of damage is LITERALLY A THING THEY CAN DO. Regardless of what healing is needed afterwards, this is ridiculous. Not as ridiculous as it would be for humans to have this ability, but that doesn't give the current state of things any good reason to be the way they are. Robos go down slow, they get up slow. I'm asking for Robos to require repairs according to the amount of damage they have recieved. In other words, I'm asking for a reasonable approach instead of compounding ridiculous developments.

While Robot not having pain mean they don't get into a pain crit, that also mean when they do drop mean they just been damaged exactly or slightly over their HP, The only instances where they get massively damaged over the 200 damage threshold is by FF. On paper, Robot is on -99 HP and got slashed by primo Rav (30 slash), its only 29 HP over the 200 limit, which is negligible considering its 2 welding cycle, what it only do is putting the burden on the medic to reserve space for a welder/cable (that don't fit white webbing), and im pretty sure defib didn't heal robot like it would heal normal human, so making it automatically go to -99HP is fine.

**Comment by Pixelknightly** [comment]
> > 
> 
> **Robos go down slow, they get up slow.**

Or is this where your misunderstanding is? The idea that robots get back up fast? Because they don't. They really, really don't. No passive healing, remember? A robot healing itself isn't the same as a human healing themselves because the human can actually fight while they do so.

Honestly, I'm just confused. You should know basically every human marine premeds. They're way tankier than any robot. (Especially with sprinting.) So the idea that robots go down slowly and get up slowly isn't really true.

Their only true advantage over human marines is the lack of long-term damage against them. Otherwise, they're worse in every aspect.

**Comment by ilikesaying** [comment]
The only way a Robot would be able to get back up fast would be if nearby robots or corpsmen/engineers decide to use their cable coils and welding torches to repair the revived robot, as repairing others is faster than repairing oneself, which is what solder used to be used for without the resource consumption.

**Comment by Pariah919** [comment]
Why not just make cryopods just accept robots, just say they heal people through nanobots.

Wait, i'm the lore master I can just say they do

**Comment by ClosetedSkeletonGH** [comment]
I still think there are a million ways of handling a robo nerf/balance that isn’t just adding human mechanics to them.

 Making them vulnerable to water, making them have different limb health balances that has their limbs come off easier (their whole hand/foot thing is messy at best), making people who don’t like robots just play a few rounds as robots so they understand how to counter them

**Comment by Sebbe9123** [comment]
> Why not just make cryopods just accept robots, just say they heal people through nanobots.
> 
> Wait, i'm the lore master I can just say they do

Yeah I'll look into something closer to this but it won't be related to cryo in the slightest. Cryo itself doesn't heal, it uses chems to heal... well except for toxins and oxy.
I guess I'll just od as tivi suggested and make a glorified locker that just locks and heals the patient after twenty seconds or so.

**Comment by QualityVan** [comment]
Cryo does actually heal on its own, but only basic damage and then slowly.

**Comment by Lumipharon** [comment]
> > Why not just make cryopods just accept robots, just say they heal people through nanobots.
> > Wait, i'm the lore master I can just say they do
> 
> Yeah I'll look into something closer to this but it won't be related to cryo in the slightest. Cryo itself doesn't heal, it uses chems to heal... well except for toxins and oxy. I guess I'll just od as tivi suggested and make a glorified locker that just locks and heals the patient after twenty seconds or so.

The ideal solution is essentially a robot version of an autodoc.
Plug in bot, repairs them, attaches new limbs if missing etc. But with the amount of code on both cryo and autodoc, its why I said its just way better to make (codewise) a new object instead of a child type with 95% of code overridden.

**Comment by Sebbe9123** [comment]
Added the machine, just missing a species flag lock and to be mapped in everywhere.

**Comment by InterroLouis** [comment]
Can we not with that power usage amount?

**Comment by Sebbe9123** [comment]
> Can we not with that power usage amount?

Yup! I'll bump it down to 10k or something, also just realized I forgot to remove the nanopaste stuff. Elsa plz map soon

**Comment by Sebbe9123** [comment]
Alright, should be ready for review now.

File: code/game/objects/items/scanners.dm [hunk 1]
```diff
@@ -107,7 +107,7 @@ REAGENT SCANNER
 	if(isxeno(M))
 		balloon_alert(user, "Unknown entity")
 		return
-	if(M.species.species_flags & NO_SCAN)
+	if(M.species.species_flags & IS_SYNTHETIC)
```
Comment (Lumipharon) [inline code/game/objects/items/scanners.dm:110]: ?? the no_scan flag is specifically to... make you not scannable.

File: code/game/objects/items/scanners.dm [hunk 2]
```diff
@@ -151,6 +151,7 @@ REAGENT SCANNER
 /obj/item/healthanalyzer/ui_static_data(mob/user)
 	var/list/data = list(
 		"patient" = patient.name,
+		"species" = patient.species.species_flags,
```
Comment (Lumipharon) [inline code/game/objects/items/scanners.dm:154]: Why is this here? It's incorrect and you just override it in the other line you added below.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+	var/repairing = FALSE
+
+	var/release_notice = TRUE
+
+	var/mob/living/carbon/human/occupant = null
+
+	var/forceeject = FALSE
+
+	var/event = 0
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:29]: autodoc.
Also those true/false ones should probably just be flags.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+	//It uses power
+	use_power = ACTIVE_POWER_USE
+	idle_power_usage = 15
+	active_power_usage = 10000 // It rebuilds you from nothing...
+
+	var/obj/item/radio/headset/mainship/doc/radio
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:36]: autodoc

File: code/game/objects/machinery/robotic_cradle.dm [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	if(occupant.stat == DEAD)
+		say("Patient has expired.")
+		repairing = FALSE
+		go_out(CRADLE_NOTICE_DEATH)
+		return
+
+	if(!repairing)
+		return
+
+/obj/machinery/robotic_cradle/proc/repair_op()
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:73]: More autodoc. Seems to be missing everywhere, I won't point them all out.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+
+/obj/machinery/robotic_cradle/process()
+	if(!occupant)
+		return
+
+	if(occupant.stat == DEAD)
+		say("Patient has expired.")
+		repairing = FALSE
+		go_out(CRADLE_NOTICE_DEATH)
+		return
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:68]: We kick out the dead robot, but repair_op() revives a dead robot?

File: code/game/objects/machinery/robotic_cradle.dm [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+
+	if(!repairing)
+		return
+
+/obj/machinery/robotic_cradle/proc/repair_op()
+
+	if(QDELETED(occupant) || occupant.stat == DEAD)
+		if(!ishuman(occupant))
+			stack_trace("Non-human occupant made its way into the autodoc: [occupant] | [occupant?.type].")
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:77]: why can humans get in?

File: code/game/objects/machinery/robotic_cradle.dm [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+		visible_message("[src] buzzes.")
+		go_out(CRADLE_NOTICE_DEATH) //kick them out too.
+		return
+
+	occupant.revive()
+	visible_message("\The [src] clicks and opens up having finished the requested operations.")
+	repairing = 0
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:84]: FALSE

File: code/game/objects/machinery/robotic_cradle.dm [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+/obj/machinery/robotic_cradle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(!ishuman(user))
+		return // no
+
+	else if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
+		var/obj/item/healthanalyzer/J = I
+		J.attack(occupant, user)
+		return
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:187]: can you even med analyse a robot? Is this all just copy paste form autodoc?

File: code/game/objects/machinery/robotic_cradle.dm [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+		to_chat(user, span_notice("[src] is compatible with humanoid anatomies only!"))
+		return
+
+	else if(M.abiotic())
+		to_chat(user, span_warning("Subject cannot have abiotic items on."))
+		return
+
+	else if(M.get_species() == "Human")
+		to_chat(user, span_warning("Subject is biological, cannot repair."))
+		return
```
Comment (Lumipharon) [inline code/game/objects/machinery/robotic_cradle.dm:230]: We're checking if not human (type path) and also if they ARE human (species)? Why not jsut do one check to see if they're a robot?

**Review by Lumipharon — CHANGES_REQUESTED** [review]
There's probably more stuff, but there seems to be a considerable amount of copy paste from old autodoc code.

If they're going to be so closely related, it probably needs to have the cradle and autodoc be children of a generic parent, with all this shared code.

File: code/game/objects/machinery/robotic_cradle.dm (hunk 7 above)
Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm:77]: They can't

File: code/game/objects/machinery/robotic_cradle.dm (hunk 9 above)
Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm:187]: Yes, this PR adds medscan to robots

File: code/game/objects/machinery/robotic_cradle.dm (hunk 10 above)
Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm:230]: Robots are human

File: code/game/objects/machinery/robotic_cradle.dm (hunk 6 above)
Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm:68]: I was thinking flavor, something something robot alive to properly fix it's CPU(cloneloss)

File: code/game/objects/items/scanners.dm (hunk 2 above)
Comment (Sebbe9123) [reply code/game/objects/items/scanners.dm]: Thought I was defining it here and making it a list or something downstairs. Fixed.

File: code/game/objects/items/scanners.dm (hunk 1 above)
Comment (Sebbe9123) [reply code/game/objects/items/scanners.dm]: Yeah it was a hack to not remove the no_scan flag from robots. fixed

File: code/game/objects/machinery/robotic_cradle.dm (hunk 10 above)
Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm:230]: Oh and there's like 10 subspecies of robot that doesn't have the same species as robot, so checking to see if the person entering is a regular Human works. All marine type species are human, but not Human if that amkes any sense. ishuman is pretty much just a check to make sure xenos don't do something

File: code/game/objects/machinery/robotic_cradle.dm [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+
+	else if(!ishuman(M)) // No monkee or beano
+		to_chat(user, span_notice("[src] is compatible with humanoid anatomies only!"))
+		return
+
+	else if(M.abiotic())
+		to_chat(user, span_warning("Subject cannot have abiotic items on."))
+		return
+
+	else if(M.get_species() == "Human")
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:228]: Unnecesarry elses here. Also the species check should be changed to `ishumanbasic(M)`.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+
+	if(!ishuman(user))
+		return // no
+
+	else if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
+		var/obj/item/healthanalyzer/J = I
+		J.attack(occupant, user)
+		return
+
+	else if(!istype(I, /obj/item/grab))
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:189]: Elses unnecesarry here too. Also you're checking if the item is a grab twice for no reason.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+
+	say("Automatic mode engaged, initialising procedure.")
+	addtimer(CALLBACK(src, PROC_REF(auto_start)), 20 SECONDS)
+
+///Callback to start auto mode on someone entering
+/obj/machinery/robotic_cradle/proc/auto_start()
+	if(repairing)
+		return
+	if(!occupant)
+		say("Occupant missing, procedures canceled.")
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:174]: Forgot a return

File: code/game/objects/machinery/robotic_cradle.dm [hunk 14]
```diff
@@ -0,0 +1,311 @@
...
+
+/obj/machinery/robotic_cradle/update_icon_state()
+	if(machine_stat & NOPOWER)
+		icon_state = "No power"
+	else if(repairing)
+		icon_state = "pod_1"
+	else if (occupant)
+		icon_state = "pod_0"
+	else
+		icon_state = "idle"
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:50]:
```suggestion
	if(machine_stat & NOPOWER)
		icon_state = "No power"
		return
	if(repairing)
		icon_state = "pod_1"
		return
	if(occupant)
		icon_state = "pod_0"
		return
	icon_state = "idle"
```

File: code/game/objects/machinery/robotic_cradle.dm (hunk 3 above)
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:29]: This var is also just not changed anywhere. Why does it exist?

File: code/game/objects/machinery/robotic_cradle.dm [hunk 15]
```diff
@@ -0,0 +1,311 @@
...
+#define CRADLE_NOTICE_DEATH 2
+#define CRADLE_NOTICE_NO_RECORD 3
+#define CRADLE_NOTICE_NO_POWER 4
+#define CRADLE_NOTICE_XENO_FUCKERY 5
+#define CRADLE_NOTICE_IDIOT_EJECT 6
+#define CRADLE_NOTICE_FORCE_EJECT 7
+//Cradle
+
+/obj/machinery/robotic_cradle
+	name = "Robotic Cradle"
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:11]:
```suggestion
	name = "robotic cradle"
```
Name of stuff is lowercase

File: code/game/objects/machinery/robotic_cradle.dm [hunk 16]
```diff
@@ -0,0 +1,311 @@
...
+	max_integrity = 350
+	soft_armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, FIRE = 30, ACID = 30)
+
+	var/locked = FALSE
+
+	var/repairing = FALSE
+
+	var/release_notice = TRUE
+
+	var/mob/living/carbon/human/occupant = null
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:25]: This will harddel.

File: code/game/objects/machinery/robotic_cradle.dm (hunk 4 above)
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:36]: This will harddel.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 17]
```diff
@@ -1195,6 +1195,7 @@
 	span_xenodanger("We suddenly feel \the [victim]'s life force streaming into us!"))
 
 	victim.do_jitter_animation(2)
+	//Gives marines less max hp.
```
Comment (ivanmixo) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1198]: Not a necessary comment

File: code/game/objects/machinery/robotic_cradle.dm [hunk 18]
```diff
@@ -0,0 +1,311 @@
...
+	say("Beginning repair procedure.")
+	repair_op()
+
+/obj/machinery/robotic_cradle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(!ishuman(user))
+		return // no
+
+	else if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:184]:
```suggestion
	if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
```

File: code/game/objects/machinery/robotic_cradle.dm [hunk 19]
```diff
@@ -0,0 +1,311 @@
...
+		var/implants = list(/obj/item/implant/neurostim)
+		var/mob/living/carbon/human/H = occupant
+		var/doc_dat
+		med_scan(H, doc_dat, implants, TRUE)
+		start_processing()
+		for(var/obj/O in src)
+			qdel(O)
+
+	say("Automatic mode engaged, initialising procedure.")
+	addtimer(CALLBACK(src, PROC_REF(auto_start)), 20 SECONDS)
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:167]:
```suggestion
	if(!do_after(dropped, 1 SECONDS, FALSE, src, BUSY_ICON_GENERIC))
		return
	if(occupant)
		to_chat(dragger, span_notice("[src] is already occupied!"))
		return
	dropped.stop_pulling()
	dropped.forceMove(src)
	occupant = dropped
	icon_state = "pod_0"
	var/implants = list(/obj/item/implant/neurostim)
	var/mob/living/carbon/human/H = occupant
	var/doc_dat
	med_scan(H, doc_dat, implants, TRUE)
	start_processing()
	for(var/obj/O in src)
		qdel(O)

	say("Automatic mode engaged, initialising procedure.")
	addtimer(CALLBACK(src, PROC_REF(auto_start)), 20 SECONDS)
```
I know autodoc does it like you did it here but autodoc code isn't the best.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 20]
```diff
@@ -0,0 +1,311 @@
...
+			if(CRADLE_NOTICE_FORCE_EJECT)
+				playsound(src.loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+				reason = "Reason for discharge: Destruction of linked CRADLE Engineering System. Alerting security advised."
+		radio.talk_into(src, "<b>Patient: [occupant] has been released from [src] at: [get_area(src)]. [reason]</b>", RADIO_CHANNEL_MEDICAL)
+	occupant = null
+	update_icon()
+	stop_processing()
+
+/obj/machinery/robotic_cradle/proc/move_inside_wrapper(mob/living/dropped, mob/dragger)
+	if(dragger.incapacitated() || !ishuman(dragger) || dragger.get_species() == "Human")
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:130]: `ishumanbasic(dragger)`

File: code/game/objects/machinery/robotic_cradle.dm [hunk 21]
```diff
@@ -0,0 +1,311 @@
...
+	if(X.status_flags & INCORPOREAL || X.do_actions)
+		return
+	visible_message(span_warning("[X] begins to pry the [src]'s cover!"), 3)
+	playsound(src,'sound/effects/metal_creaking.ogg', 25, 1)
+	if(!do_after(X, 2 SECONDS))
+		return
+	playsound(loc, 'sound/effects/metal_creaking.ogg', 25, 1)
+	go_out()
+
+/obj/machinery/robotic_cradle/proc/go_out(notice_code = FALSE)
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:100]: Doc

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Also doc all vars on the type and every new proc you define.

File: code/game/objects/machinery/robotic_cradle.dm [hunk 22]
```diff
@@ -0,0 +1,322 @@
...
+	//This var is used to stop more robits from being shoved inside.
+	var/locked = FALSE
+	//This var is used to see if the machine is currently repairing or not.
+	var/repairing = FALSE
+	//This var is the reference used to pick what notification medics get when a patient is released.
+	var/release_notice = TRUE
+	//This var is the reference used for the patient
+	var/mob/living/carbon/human/occupant = null
+	//This var is used to eject the robot forcefully, dealing damage to them.
+	var/forceeject = FALSE
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:27]: Not used anymore

File: code/game/objects/machinery/robotic_cradle.dm [hunk 23]
```diff
@@ -0,0 +1,322 @@
...
+		return
+
+	if(!istype(I, /obj/item/grab))
+		return
+
+	if(machine_stat & (NOPOWER|BROKEN))
+		to_chat(user, span_notice("[src] is non-functional!"))
+		return
+
+	else if(occupant)
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:209]:
```suggestion
	if(occupant)
```

**Review by ivanmixo — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/robotic_cradle.dm [hunk 24]
```diff
@@ -137,7 +135,7 @@
 
 //This proc is what a robot calls when they try to enter a cradle on their own.
 /obj/machinery/robotic_cradle/proc/move_inside_wrapper(mob/living/dropped, mob/dragger)
-	if(dragger.incapacitated() || !ishuman(dragger) || ishumanbasic(dragger))
+	if(dragger.incapacitated() || !isrobot(dropped) || ishumanbasic(dragger))
```
Comment (ivanmixo) [inline code/game/objects/machinery/robotic_cradle.dm:138]: You shouldve turned the other one into that check. Now non humans can't put robots in.

Comment (Sebbe9123) [reply code/game/objects/machinery/robotic_cradle.dm]: Sorry, sleepy. fixed

**Review by ivanmixo — APPROVED** [review]

**Comment by ivanmixo** [comment]
Eh uh last few nitpicks.

The cradle doesn't have a tile under it on pillar of spring and it feels randomly placed just in the middle of medbay.

The cradle doesn't have a description.

**Comment by Sebbe9123** [comment]
> Eh uh last few nitpicks.
> 
> The cradle doesn't have a tile under it on pillar of spring and it feels randomly placed just in the middle of medbay.
> 
> The cradle doesn't have a description.

Done, didn't notice neither the desc or the plating. Mapping isn't my strong suit :P

**Review by Kuro020 — CHANGES_REQUESTED** [review]
Coder sprite begone

**Comment by AusarAtkin** [comment]
Does sleep can also heal cloneloss on robot? Might as well check on it to make it fair for em to have on the field solution

**Comment by Sebbe9123** [comment]
> Coder sprite begone

Please leave a review with actual content, this isn't useful feedback at all for anyone ever.

**Review by Kuro020 — APPROVED** [review]
